src/Lykke.Service.ExchangeDataStore/Lykke.Service.ExchangeDataStore.AzureRepositories/OrderBooks/OrderBookSnapshotsRepository.cs

```diff
@@ -0,0 +1,90 @@
+﻿using AzureStorage;
+using Common.Log;
+using Lykke.Service.ExchangeDataStore.Core.Domain;
+using Lykke.Service.ExchangeDataStore.Core.Domain.OrderBooks;
+using Lykke.Service.ExchangeDataStore.Core.Settings.ServiceSettings;
+using Newtonsoft.Json;
+using Polly;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Lykke.Service.ExchangeDataStore.AzureRepositories.OrderBooks
+{
+    // ReSharper disable once ClassNeverInstantiated.Global - autofac injected
+    internal class OrderBookSnapshotsRepository : IOrderBookSnapshotsRepository
+    {
+        private readonly string _blobContainer;
+        private static readonly string _className = nameof(OrderBookSnapshotsRepository);
+
+        private readonly ILog _log;
+        private readonly INoSQLTableStorage<OrderBookSnapshotEntity> _tableStorage;
+        private readonly IBlobStorage _blobStorage;
+        private readonly JsonSerializerSettings _serializerSettings = new JsonSerializerSettings
+        {
+            DateFormatString = Constants.JsonSerializationTimestampFormat
+        };
+
+        public OrderBookSnapshotsRepository(INoSQLTableStorage<OrderBookSnapshotEntity> tableStorage, IBlobStorage blobStorage, ILog log, AzureTableConfiguration dbConfig)
+        {
+            _tableStorage = tableStorage;
+            _blobStorage = blobStorage;
+            _log = log;
+            _blobContainer = dbConfig.EntitiesBlobContainerName;
+        }
+
+        public async Task SaveAsync(IOrderBookSnapshot orderBook)
```

Comment:
I would pass here a cancellation token. There is no way to stop this method if the service is about to stop

Reply:
Done.